res/skins/QMLDemo/util.mjs

```diff
@@ -0,0 +1,8 @@
+let previous = undefined;
+// primitive function to check whether a value changed
+// between invocations
+export function hasChanged(value) {
+    const changed = value != previous;
+    previous = value;
+    return changed;
+}
```

Comment:
Why is this a utility function? If I use this function for two unrelated values, the result is wrong, so it's not reusable.

Reply:
Yes, it is very primitive, I'm aware. It's not supposed to be perfect but just generally make sure that code that gets called repeatedly can debounce a bit so to speak.